### ISSUE TITLE
[develop 1.2.2] - error message bad_index_check()

### ISSUE THREAD

**Issue by pautiina** [Cacti/cacti#2383]
After update - an error is displayed:
```
2019/02/08 09:46:40 - POLLER: Poller[Main Poller] WARNING: You have Array Devices with bad SNMP Indexes. Devices: Device[ BURLUK-CORE-MIKROT], Device[LIPCI-PON], Device[DNS], Device[10.0.3.23] totalling 38 Data Sources. Please Either Re-Index, Delete or Disable these Data Sources.
--
2019/02/08 09:46:40 - CMDPHP PHP ERROR NOTICE Backtrace: (/poller.php[777]:bad_index_check(), /poller.php[799]:CactiErrorHandler())
2019/02/08 09:46:40 - ERROR PHP NOTICE: Array to string conversion in file: /var/www/cacti/poller.php on line: 799
```
![image](https://user-images.githubusercontent.com/11805503/52465359-0a9ab080-2b87-11e9-8d28-b425b360ca5c.png)
After the recommendations of the poller and the removal of the orphaned graphs - the error disappeared

**Comment by cigamit** [comment]
I hate it when that happens.  You think you have checked the error three times.  All works and then you commit, and it's not working.

So, after you update lib/data_query.php, lib/functions.php, lib/template.php and data_sources.php, and include/global_settings.php, re-index the devices in question.  Then goto Console > Management > Data Sources and pick Status = Bad Index.

I'll look into the message now.  It may be as simple as I forgot to commit something.

**Comment by cigamit** [comment]
Message fixed.  This is a new periodic warning that will be introduced in 1.2.2 that allows users to track down bad indexes.  That is a graph had a data source with an index, and that index disappeared.  When this happens, the data source will be removed from the poller cache.  Upon subsequent re-index, if the index re-appears, the graph will pick right back up graphing.

I did a major audit of of re-indexing for 1.2.0, but obviously got a few things wrong.  At least they were not catastrophic well at least not long term.

I wanted to provide users the ability to more easily diagnose these types of problems.  Goal make it simpler.  I did not want to go as far as create a FixIt button as the developers here will never know the full dimensions of a specific users problems until they are experienced.  So, for now, we just added the new Status.
